great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -918,43 +927,12 @@ def resolve_metric_bundle(  # noqa: C901 #  too complex
 
         res: List[sqlalchemy.Row]
 
-        # We need a different query for each Domain (where clause).
-        queries: dict[IDDictID, dict] = {}
-
-        query: dict
-
-        domain_id: IDDictID
-
-        bundled_metric_configuration: MetricComputationConfiguration
-        for bundled_metric_configuration in metric_fn_bundle:
-            metric_to_resolve: MetricConfiguration = (
-                bundled_metric_configuration.metric_configuration
-            )
-            metric_fn: Any = bundled_metric_configuration.metric_fn
-            compute_domain_kwargs: dict = bundled_metric_configuration.compute_domain_kwargs or {}
-            if not isinstance(compute_domain_kwargs, IDDict):
-                compute_domain_kwargs = IDDict(compute_domain_kwargs)
-
-            domain_id = compute_domain_kwargs.to_id()
-            if domain_id not in queries:
-                queries[domain_id] = {
-                    "select": [],
-                    "metric_ids": [],
-                    "domain_kwargs": compute_domain_kwargs,
-                }
-
-            if self.engine.dialect.name == "clickhouse":
-                queries[domain_id]["select"].append(
-                    metric_fn.label(
-                        metric_to_resolve.metric_name.join(
-                            random.choices(string.ascii_lowercase, k=4)
-                        )
-                    )
-                )
-            else:
-                queries[domain_id]["select"].append(metric_fn.label(metric_to_resolve.metric_name))
-
-            queries[domain_id]["metric_ids"].append(metric_to_resolve.id)
+        queries: dict[IDDictID, dict] = self._organize_metrics_by_domain(
```

Comment:
One last more minor comment. I find the fact that we're returning a dict here confusing. A dict implies that the caller would be looking something up. They're not, they're just using the list of values. We should return a list from this method to show that there's nothing to look up (the IDs are essentially made up by the method)